src/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -323,6 +323,12 @@ function VisEditor($rootScope, $scope, $route, timefilter, AppState, $window, kb
       if (id) {
         notify.info('Saved Visualization "' + savedVis.title + '"');
         if ($scope.isAddToDashMode()) {
+          // We need to manually set the last url for this app so navigating back to visualize will open the saved
+          // visualization, not the unsaved one.
+          const savedVisualizationUrl =
+            kbnUrl.eval(`${kbnBaseUrl}#${VisualizeConstants.EDIT_PATH}/{{id}}`, { id: savedVis.id });
+          chrome.trackSubUrlForApp('kibana:visualize', savedVisualizationUrl);
```

Comment:
The main reason I like this more is that it is continuing to do navigation like the rest of the system expects. I'm open to other ideas, but I think that modifying the last sub url of a tab is really crossing into chrome's workspace and forcing visualize to care a lot about how chrome accomplishes things. Either way, I guess it's sort of like that... let me think about it for a bit...